src/main/java/org/kuali/kfs/krad/config/KRADConfigurer.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * The Kuali Financial System, a comprehensive financial management system for higher education.
+ *
+ * Copyright 2005-2016 The Kuali Foundation
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.kuali.kfs.krad.config;
+
+import org.kuali.kfs.krad.service.DataDictionaryService;
+import org.kuali.kfs.krad.service.KRADServiceLocatorInternal;
+import org.kuali.kfs.krad.service.KRADServiceLocatorWeb;
+import org.kuali.kfs.krad.util.KRADConstants;
+import org.kuali.rice.core.api.config.module.RunMode;
+import org.kuali.rice.core.api.config.property.ConfigContext;
+import org.kuali.rice.core.api.resourceloader.GlobalResourceLoader;
+import org.kuali.rice.core.framework.config.module.ModuleConfigurer;
+import org.springframework.context.ApplicationEvent;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.context.event.SmartApplicationListener;
+
+import javax.sql.DataSource;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class KRADConfigurer extends ModuleConfigurer implements SmartApplicationListener {
+
+    private DataSource applicationDataSource;
+
+
+    private static final String KRAD_SPRING_BEANS_PATH = "classpath:org/kuali/kfs/krad/config/KRADSpringBeans.xml";
+    private static final String KNS_SPRING_BEANS_PATH = "classpath:org/kuali/kfs/kns/config/KNSSpringBeans.xml";
+    private static final String CORE_SERVICE_SPRING_BEANS_PATH = "classpath:org/kuali/kfs/coreservice/config/CoreServiceLocalSpringBeans.xml";
+
+    public KRADConfigurer() {
+        // TODO really the constant value should be "krad" but there's some work to do in order to make
+        // that really work, see KULRICE-6532
+        super(KRADConstants.KR_MODULE_NAME);
+        setValidRunModes(Arrays.asList(RunMode.LOCAL));
+    }
+
+    @Override
+    public void addAdditonalToConfig() {
+        configureDataSource();
+    }
+
+    @Override
+    public List<String> getAdditionalSpringFiles() {
+        final String files = ConfigContext.getCurrentContextConfig().getProperty("kfs." + getModuleName() + ".additionalSpringFiles");
```

Comment:
This seems like something other institutions would also have to do, can it be included in the kuali co version of the code in the future?